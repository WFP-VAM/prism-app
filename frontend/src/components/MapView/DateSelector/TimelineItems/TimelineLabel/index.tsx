import {
  WithStyles,
  createStyles,
  withStyles,
  Typography,
} from '@material-ui/core';
import React from 'react';
import { DateRangeType } from '../../../../../config/types';
import { moment } from '../../../../../i18n';
import { MONTH_YEAR_DATE_FORMAT } from '../../../../../utils/name-utils';

const TimelineLabel = ({ classes, locale, date }: TimelineLabelProps) => {
  if (date.isFirstDay) {
    return (
      <Typography variant="body2" className={classes.dateItemLabel}>
        {moment(date.value).locale(locale).format(MONTH_YEAR_DATE_FORMAT)}
      </Typography>
    );
  }
  return <div className={classes.dayItem} />;
};

const styles = () =>
  createStyles({
    dateItemLabel: {
      color: '#101010',
      position: 'absolute',
      top: 33,
      textAlign: 'left',
      paddingLeft: 2,
      minWidth: 400,
      fontWeight: 'bold',
      zIndex: 1,
    },

    dayItem: {
      height: 10,
    },
  });

export interface TimelineLabelProps extends WithStyles<typeof styles> {
  locale: string;
  date: DateRangeType;
}

export default withStyles(styles)(TimelineLabel);
