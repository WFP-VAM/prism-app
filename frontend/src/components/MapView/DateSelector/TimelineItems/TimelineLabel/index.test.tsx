import { render } from '@testing-library/react';
import TimelineLabel, { TimelineLabelProps } from '.';

test('TimelineLabel renders as expected', () => {
  // Arrange
  const props: Omit<TimelineLabelProps, 'classes'> = {
    locale: 'fr',
    date: {
      value: 1640883600000,
      label: '31 Dec 2021',
      month: 'Dec 2021',
      isFirstDay: true,
      date: '2021-12-31',
    },
  };

  // Act
  const { container } = render(<TimelineLabel {...props} />);

  // Assert
  expect(container).toMatchSnapshot();
});
