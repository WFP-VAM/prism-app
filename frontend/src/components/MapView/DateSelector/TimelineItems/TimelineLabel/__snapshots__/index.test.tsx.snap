// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TimelineLabel renders as expected 1`] = `
<div>
  <mock-typography
    classname="TimelineLabel-dateItemLabel-1"
    variant="body2"
  >
    d√©cembre 2021
  </mock-typography>
</div>
`;
